src/PostgREST/Utils.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE BangPatterns  #-}
+{-# LANGUAGE MagicHash     #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+module PostgREST.Utils (
+  recursiveSizeNF
+) where
+
+import Control.Monad
+
+import GHC.Exts
+import GHC.Exts.Heap           hiding (size)
+import GHC.Exts.Heap.Constants (wORD_SIZE)
+
+import System.Mem
+
+import Protolude
+
+-- Code in this module is taken from:
+--    https://hackage.haskell.org/package/ghc-datasize-0.2.7
```

Comment:
@taimoorzaeem Could you remind me why we need to vendor this code here? Maybe it's possible to use the dependency directly?

Reply:
https://github.com/PostgREST/postgrest/pull/3801#discussion_r1857313684

Reply:
Adding the dependency directly would have been perfect. The only issue is that `ghc-datasize-0.2.7` is marked broken in nixpkgs.

```bash
error: Package ‘ghc-datasize-0.2.7’ in /nix/store/hyy4vjyamr7pj1br9y8r1ssssqp570y2-source/pkgs/development/haskell-modules/hackage-packages.nix:120611 is marked as broken, refusing to evaluate.

       a) To temporarily allow broken packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_BROKEN=1
```

Reply:
Works! Thanks.